src/device-detector.service.ts

```diff
@@ -23,8 +24,11 @@ export class DeviceDetectorService {
     device = '';
     os_version = '';
     browser_version = '';
-    constructor() {
-        this.ua = window.navigator.userAgent;
+
+    constructor(@Inject(PLATFORM_ID) private platformId) {
+        if (isPlatformBrowser(platformId)) {
```

Comment:
@staticint please try `if (isPlatformBrowser(this.platformId))` instead to fix the tslint error.

Reply:
done